subsys/bluetooth/host/sdp.c

```diff
@@ -2334,57 +2362,64 @@ static int sdp_get_uuid_data(const struct bt_sdp_attr_item *attr,
 		left = slen;
 
 		/* check if at least DTD + UUID16 can be read safely */
-		if (left < 3) {
+		if (left < (sizeof(dtd) + BT_UUID_SIZE_16)) {
 			return -EMSGSIZE;
 		}
 
 		/* check DTD and get stacked UUID value */
-		switch (p[0]) {
+		dtd = p[0];
```

Comment:
Below we increment `p` with `++p` before getting the UUIDs. I guess it would make more sense to increment it here where we extract the `dtd` value, and then remove the increment in lines 2374 and 2386.

Reply:
Updated, Thanks. Move left -= sizeof(dtd); to this line too.